numpy/_core/records.py

```diff
@@ -238,18 +238,17 @@ def __getattribute__(self, attr):
                     "attribute '%s'" % attr)
 
     def __setattr__(self, attr, val):
-        if attr in ('setfield', 'getfield', 'dtype'):
+        if attr in {'setfield', 'getfield', 'dtype'}:
             raise AttributeError("Cannot set '%s' attribute" % attr)
         fielddict = nt.void.__getattribute__(self, 'dtype').fields
         res = fielddict.get(attr, None)
         if res:
             return self.setfield(val, *res[:2])
+        elif getattr(self, attr, None):
```

Comment:
some other ruff rule will complain about this later anyway, so might as well do

```suggestion
        if getattr(self, attr, None):
```

Reply:
If you don't mind, I'd rather wait for that rule to be enabled and apply fixes automatically when feasible.